frontend/src/utils/CampUtils.ts

```diff
@@ -159,3 +205,88 @@ export const getSessionBorderColor = (sessionIndex: number): string =>
 
 export const getSessionFillColor = (sessionIndex: number): string =>
   FILL_COLORS[sessionIndex % 8];
+
+// isEDLPValid checks if the EDLP config in the request object is valid
+const isEDLPValid = (camp: CreateUpdateCampRequest): boolean => {
+  // Camp either doesn't have EDLP or has both
+  if (
+    (camp.earlyDropoff && !camp.latePickup) ||
+    (camp.latePickup && !camp.earlyDropoff)
+  ) {
+    return false;
+  }
+  // Camp must have both late pickup and early dropoff fee if it has EDLP
+  if (
+    camp.earlyDropoff &&
+    camp.latePickup &&
+    (!camp.dropoffFee || !camp.pickupFee)
+  ) {
+    return false;
+  }
+  return true;
+};
+
+// isMinCampDetailsFilled checks if the camp object has the minimum details required to create/update a camp
+export const isMinCampDetailsFilled = (
+  camp: CreateUpdateCampRequest,
+): boolean => {
+  if (
+    camp.name &&
+    camp.description &&
+    camp.fee &&
+    camp.startTime &&
+    camp.endTime &&
+    camp.ageLower &&
+    camp.ageUpper &&
+    isEDLPValid(camp) &&
+    camp.location.streetAddress1 &&
+    camp.location.city &&
+    camp.location.province &&
+    camp.location.province &&
+    camp.location.postalCode &&
+    camp.campSessions.length > 0 &&
+    camp.campSessions.every((cs) => cs.capacity > 0)
+  ) {
+    return true;
+  }
+  return false;
+};
+
+// createUpdateCamp creates or updates a camp with the given fields
+export const createUpdateCamp = async (
```

Comment:
I think this belongs in CampsAPIClient, but I also don't see the huge benefit of having this as a combined method, since so much logic is based on the `isNewCamp` boolean

Reply:
just wanted to reduce code duplication since the code is pretty much the same but the text and call is conditionally rendered